frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,255 @@
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+import {
+  NamespaceModel,
+  PersistentVolumeClaimModel,
+  VolumeSnapshotModel,
+  VolumeSnapshotClassModel,
+} from '@console/internal/models';
+import { getName, getNamespace, Status } from '@console/shared';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+
+const dropdownUnits = {
+  Mi: 'MiB',
+  Gi: 'GiB',
+  Ti: 'TiB',
+};
+
+export const RestorePVCModal = withHandlePromise((props: RestorePVCModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress, handlePromise } = props;
+  const [restorePVCName, setPVCName] = React.useState(`${getName(resource) || 'pvc'}-restore`);
+  const defaultSize: string[] = resource?.status?.restoreSize
+    ? validate.split(resource?.status?.restoreSize)
+    : null;
+  const pvcRequestedSize = resource?.status?.restoreSize
+    ? `${defaultSize[0]} ${dropdownUnits[defaultSize[1]]}`
+    : '';
+  const [requestedSize, setRequestedSize] = React.useState(defaultSize ? defaultSize[0] : '');
+  const [requestedUnit, setRequestedUnit] = React.useState(defaultSize ? defaultSize[1] : 'Gi');
+  const [pvcSC, setPVCStorageClass] = React.useState<string>(null);
+  const [progress, setProgress] = React.useState(true);
+  const [driver, setDriver] = React.useState<string>(null);
+  const [pvcData, setPVCData] = React.useState<PersistentVolumeClaimKind>(null);
+  const namespace = getNamespace(resource);
+
+  React.useEffect(() => {
+    Promise.all([
+      k8sGet(
+        PersistentVolumeClaimModel,
+        resource?.spec?.source?.persistentVolumeClaimName,
+        namespace,
+      ).catch((e) => {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching objects', e); // Not important to show user
+        setProgress(false);
+      }),
+      k8sGet(VolumeSnapshotClassModel, resource.spec.volumeSnapshotClassName),
+    ])
+      .then((obj) => {
+        setPVCData(obj?.[0]);
+        setDriver(obj?.[1]?.driver);
+        setProgress(false);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching objects', e); // Not important to show user
+        setProgress(false);
+      });
+  }, [namespace, resource]);
+
+  const onlyPvcSCs = (scObj: StorageClassResourceKind) => {
+    if (driver) {
+      return scObj.provisioner.includes(driver);
+    }
+    return false;
+  };
+
+  const requestedSizeInputChange = ({ value, unit }) => {
+    setRequestedSize(value);
+    setRequestedUnit(unit);
+  };
+
+  const handleStorageClass = (updatedStorageClass) => {
+    const scName: string = updatedStorageClass?.metadata?.name || '';
+    setPVCStorageClass(scName);
+  };
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const snapshotName = getName(resource);
+    const accessModes = pvcData?.spec?.accessModes || ['ReadWriteOnce'];
```

Comment:
```suggestion
    const accessModes = pvcData?.spec.accessModes || ['ReadWriteOnce'];
```

Reply:
can we change the access Mode for the restored PVC?

Reply:
We should not be able to, also, since `accessMode` is required field in PVC, I will remove the default value here.